### PR TITLE
Add build_optimizers

### PR DESCRIPTION
**Pull request by zengwang430521**
Add the function to bulid optimizers for sub-models in a model. This function is used in adversarial training. The function and test code are copied from mmediting.

### CODE REVIEW

File: tests/test_optimizer.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import torch
+import torch.nn as nn
+
+from mmpose.core import build_optimizers
+
+
+class ExampleModel(nn.Module):
+
+    def __init__(self):
+        super(ExampleModel, self).__init__()
```
Comment (innerlee) [inline tests/test_optimizer.py:16]: `super().__init__()`

File: tests/test_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# ------------------------------------------------------------------------------
+# Copied from https://github.com/open-mmlab/mmediting
```
Comment (innerlee) [inline tests/test_optimizer.py:2]: although it is reasonable and okay, not sure if it is a good idea to start this notice within the mm-series.

Comment (innerlee) [reply tests/test_optimizer.py:2]: cc. @hellock , we can borrow code freely in mm-*, right?

Comment (zengwang430521) [reply tests/test_optimizer.py:2]: OK, I will drop this notice.

File: mmpose/core/optimizer/builder.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+from mmcv.runner import build_optimizer
+
+
+def build_optimizers(model, cfgs):
+    """Build multiple optimizers from configs.
```
Comment (innerlee) [inline mmpose/core/optimizer/builder.py:5]: is this needed because of "multiple optimizers"?

Comment (zengwang430521) [reply mmpose/core/optimizer/builder.py:5]: Yes. The training of HMR model involves adversarial training and this needs individual optimizers for generator and discrinimator.

**Review by innerlee — APPROVED** [review]

File: tests/test_optimizer.py (hunk 2 above)
Comment (hellock) [reply tests/test_optimizer.py]: Yes we can use codes from other codebases in OpenMMLab without this notice.

**Pull request merged**